src/OSPSuite.Core/Domain/ObjectPath.cs

```diff
@@ -146,6 +149,15 @@ public virtual T Resolve<T>(IEntity refEntity) where T : class
          return resolvePath<T>(dependentObject, usePath);
       }
 
+      private void removeMatchingParentPath(IContainer root, List<string> usePath)
+      {
+         var matchingElements = !string.IsNullOrEmpty(root?.ParentPath?.PathAsString) &&
```

Comment:
same `root?.ParentPath?.PathAsString` locally in a variable  and check that it's not null or empty then return